apps/meteor/ee/server/apps/communication/rest.js

```diff
@@ -285,6 +416,63 @@ export class AppsRestApi {
 			},
 		);
 
+		this.api.addRoute(
+			'buildExternalAppRequest',
+			{ authRequired: true },
+			{
+				async get() {
+					if (!this.queryParams.appId) {
+						return API.v1.failure({ error: 'Invalid request. Please ensure an appId is attached to the request.' });
+					}
+
+					const baseUrl = orchestrator.getMarketplaceUrl();
+					const workspaceId = settings.get('Cloud_Workspace_Id');
+
+					const requester = {
+						id: this.user._id,
+						username: this.user.username,
+						name: this.user.name,
+						nickname: this.user.nickname,
+						emails: this.user.emails.map((e) => e.address),
+					};
+
+					let admins = [];
+					try {
+						const adminsRaw = await UsersRaw.findUsersInRoles('admin', undefined, {
+							projection: {
+								username: 1,
+								name: 1,
+								emails: 1,
+								nickname: 1,
+							},
+						}).toArray();
+
+						admins = adminsRaw.map((a) => {
+							return {
+								id: a._id,
+								username: a.username,
+								name: a.name,
+								nickname: a.nickname,
+								emails: a.emails.map((e) => e.address),
+							};
+						});
+					} catch (e) {
+						orchestrator.getRocketChatLogger().error('Error getting the admins to request an app be installed:', e);
+					}
+
+					const queryParams = new URLSearchParams();
+					queryParams.set('workspaceId', workspaceId);
+					queryParams.set('frameworkVersion', appsEngineVersionForMarketplace);
+					queryParams.set('requester', Buffer.from(JSON.stringify(requester)).toString('base64'));
+					queryParams.set('admins', Buffer.from(JSON.stringify(admins)).toString('base64'));
```

Comment:
are we ok on sharing these admin's data to any logged in user?

Reply:
@graywolf336

Reply:
As far as I understand we need this data to send rocket.cat notifications and end-users are the one triggering this call, so they don't have further permissions. But I'll let Mr. Brad talk about this since he is the one that wrote this part of the code.

Reply:
changed per our internal discussion 👍